### DISCUSSION TITLE
Apple Sillicon MPS NaN error

### DISCUSSION THREAD

**Discussion by blimjoe** [openai/whisper#2693] [General]
When using large model/medium model on Apple Sillicon with MPS enabled, it's truely accelerated with full almost full gpu usage. But it will crash in few mins, due to error: 

> MPS numerical instability detected: Expected parameter logits (Tensor of shape (1, 51865)) of distribution Categorical(logits: torch.Size([1, 51865])) to satisfy the constraint IndependentConstraint(Real(), 1), but found invalid values:
> tensor([[nan, nan, nan,  ..., nan, nan, nan]], device='mps:0'

It seems to be a Apple Sillicon problem? Because of the deep Transformer layer? Or cumulative errors caused by insufficient mixed-precision support in Apple Silicon GPU MPS?

**Comment by blimjoe** [comment]
But it works perfect on turbo model. WHY?

**Comment by janngobble** [comment]
I know this is an old post but I can reproduce this with OpenAI Whisper 20250625, Python 3.12.13, PyTorch 2.12.1, Apple Silicon MPS. medium and medium.en fail with NaN logits on MPS, while base.en completes. This appears separate from the --word_timestamps float64 issue in timing.py.

More info to come.
